internal/csi-addons/rbd/replication.go

```diff
@@ -724,11 +724,16 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 		log.DebugLog(ctx, "image %s, savedImageTime=%v, currentImageTime=%v", rbdVol, st, creationTime)
 
 		syncInfo, sErr := localStatus.GetLastSyncInfo(ctx)
-		if sErr != nil {
+		if sErr != nil && !errors.Is(sErr, rbderrors.ErrLastSyncTimeNotFound) {
 			return nil, status.Errorf(codes.Internal, "failed to get last sync info: %s", sErr.Error())
 		}
 
-		if req.GetForce() && st.Equal(*creationTime) && !syncInfo.IsSyncing() {
+		// consider the image is not syncing if either
+		// - the last sync time was not found or
```

Comment:
please add a comment that when last sync time is not found for example when in split brain state etc. it will help us in future why we used this check.

Reply:
> > I've added example as when images on both peered clusters are in secondary state.
> 
> I think this is wrong, because if the images are secondary state on both cluster, both will be in up+unknown state (we have check for this even before this state). This is more of an error state when not in healthy secondary state. when both are in secondary state its not possible to resync as there is no primary image.

The behavior of resync execution when both images are secondary is not changed by this pr.